torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -699,6 +700,16 @@ def view_avoid_dupes_with_primals(t):
             grad_enabled_mutation=grad_enabled_mutation,
             tokens=mode._tokens,
         )
+        # AOTAutograd collect metadata will do fake tensor propagation, but it
+        # throws out all the resulting fake tensors and doesn't save anything
+        # about sizes (TODO: Actually, the subclass metadata does save size
+        # info, this is likely to be incorrect if unbacked SymInts are
+        # allowed).  The net effect is we generate a bunch of fresh unbacked
+        # symbols that we immediately throw out and don't use.  NB: we don't
+        # want to rename into these symbols, because we aren't going to have
+        # binding sites for them.
+        if (fake_mode := detect_fake_mode()) and fake_mode.shape_env:
+            fake_mode.shape_env.pending_fresh_unbacked_symbols.clear()
```

Comment:
I think your issue here is slightly different, because you don't have any unbacked symbols in your situation, but this code path only ever is load bearing when there are unbacked symbols. The TODO here was simply me thinking "hmm, can these unbacked symbols leak out" and the answer is, only if there is a size flowing out of forward analysis.

Reply:
Ah ok I think I'm in agreement - the issue is pretty independent, but reading this change just made me realize that we'll need to do some more work for https://github.com/pytorch/pytorch/issues/124619 if we want to be able to handle "graph output is a tensor subclass with unbacked symints"

Reply:
Yes, fortunately, in our use cases the unbacked symint typically evaporates before you get out of the model. But this is going to be its own bug farm.